src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -107,7 +111,7 @@ object UmiConsensusCaller {
     rg.setDescription     (collapse(_.getDescription))
     rg.setLibrary         (collapse(_.getLibrary))
     rg.setSample          (collapse(_.getSample))
-    rg.setPlatform        (collapse(_.getPlatform))
+    rg.setPlatform        (collapse(_.getPlatform, toUpper = true))
```

Comment:
I think you can get rid of your modifications above and just inline this (since you only do it once).  My philosophy only generalize if you need to do it 2+ times.

```suggestion
    rg.setPlatform        (collapse(_.getPlatform.toUpper))  // NB: this to ensure that platforms are all upper-case; not all tools are modern or well-behaved
```

Feel free to remove my snarky comment.

Reply:
This approach fails if any platform is null. The only reason I just caught that is because its an undocumented  part of the added unit test.  

There are actually 3 read groups that are created.  The original RG as part of the `SamBuilder` class, and the two for testing purposes.  We already filter out null values as part of the `collapse` function.  

I tried this as a solution:

```scala
rg.setPlatform        (collapse(_.getPlatform.filter(_ != null).toUpperCase)) // NB: this to ensure that platforms are all upper-case; not all tools are modern or well-behaved
 ```

but I that also failed the unit test.

I can keep on plugging away if we really want to don't want parameterize the call.

Reply:
`   rg.setPlatform        (collapse(x => Option(x.getPlatform).map(_.toUpperCase).orNull))`